### PR TITLE
Restore libhandy support within Glade

### PR DESCRIPTION
**Pull request by Eonfge**
@bilelmoussaoui 

I previously made a pull request for Glade 3.38 under #14, but @xjuan decided to directly update Glade to 3.38.1

In this update, an error was introduced. While previously, Glade had support for libhandy components, the current live version does not. This was properly addressed in pull request #14, which I then later force-pushed in a last attempt to synchronize the changes. That didn't work out, so here I have a new pull request to re-include libhandy support.

### CODE REVIEW

**Comment by Eonfge** [comment]
Btw, the patch is not mine, it's an upstream commit that didn't make it into the final 1.0.0 release.
https://gitlab.gnome.org/GNOME/libhandy/-/commit/65ff089b7f346f3809e9e0050ea32ac003465ede

**Comment by A6GibKm** [comment]
bot, build org.gnome.Glade

**Comment by xjuan** [comment]
Hello guys, I do not want to include libhandy directly in the flatpak but I do want to provide a way for install extra modules
I think the best approach would be to create an extension in Glade for modules and plugins.

I never got the time to try it, @Eonfge do you?

https://blog.tingping.se/2018/03/18/flatpaking-plugins.html

**Comment by bilelmoussaoui** [comment]
I don't think that's the right approach to the problem, how many extensions there should be? imho I would just package the most "famous" libraries catalogues and call it a day.

File: org.gnome.Glade.json [hunk 1]
```diff
@@ -36,6 +36,28 @@
...
+                "-Dexamples=false",
+                "-Dglade_catalog=enabled",
+                "-Dintrospection=disabled",
+                "-Dtests=false",
+                "-Dvapi=false"
+            ],
+            "sources" : [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libhandy/1.0/libhandy-1.0.0.tar.xz",
```
Comment (bilelmoussaoui) [inline org.gnome.Glade.json]: Bump it to the latest release & remove the patch

**Comment by xjuan** [comment]
I think there should be as many extension as distinct modules so the user can install only the extensions its interested in.
Otherwise how many libraries should be bundled?

The only other big module I am aware of is libreoffice, but I do not think it should be included in Glade but they still need to be able to use it, and having an extension would be useful for everyone.

Of course we could create an extension with more than one module if that makes sense.
is there any other common library used together with libhandy?

**Comment by albfan** [comment]
totally agreed with @xjuan. This needs to be reworked into an extension.

File: org.gnome.Glade.json (hunk 1 above)
Comment (Eonfge) [reply org.gnome.Glade.json]: Updated

**Comment by Eonfge** [comment]
@xjuan @albfan You're putting the card before the horse. Let's focus on first restoring application functionality that broke 1.5 months ago. Then you can rework the system to use extensions.

**Comment by albfan** [comment]
@Eonfge because someone decide to include that without asking us.

You can build locally and have your libhandy functionality, but this should be done as an extension.

Of course we can merge this and forget, but is not the proper way, just the "I want this button in red because whatever and you have only to accept my MR". Well no, this is the flatpaked version of Glade, support libhandy is a different thing (and as any other thing will consume time and effort in the long term). At some point this things byte you in the ass as this is not anymore the vanilla version of Glade we released.

See I'm a user in other apps and have to walk this route, https://gitlab.gnome.org/GNOME/gedit/-/issues/301 As maintainer, you want to control your project, libhandy should control itself, and as a flatpak, that is to create a Glade extension. 

Probably @Kekun could give us other arguments

*NOTE*: Create a flatpak extension is not that hard :wink: 
*NOTE2*: I'm ok if @xjuan allows this to be merged, but arguments are solid.

**Comment by bilelmoussaoui** [comment]
The number of libraries that offers a glade catalogue (other than libreoffice) is libhandy/gtksourceview/webkit2gtk. The webkit2gtk case is handled by the GNOME SDK. The discoverability of extensions is not very on point and the usability of glade isn't on top neither so let's not make things harder for developers today to use the flatpaked version of Glade and figure out something that would work properly in a separate issue instead of blocking this for a month.  
Also note that's I merged libhandy support because I maintained Glade here on Flathub for a while and kept it up to date.

Creating an extension requires a PR here to be merged along with an other one in flathub/flathub.

**Comment by A6GibKm** [comment]
[Let](https://gitlab.gnome.org/World/PasswordSafe/) [me](https://gitlab.gnome.org/msandova/fractal) [point](https://gitlab.gnome.org/haecker-felix/souk) [out](https://gitlab.gnome.org/World/Shortwave) [that](https://gitlab.gnome.org/World/Social) [every](https://gitlab.gnome.org/World/Fragments) [single](https://gitlab.gnome.org/bilelmoussaoui/authenticator) [modern](https://gitlab.gnome.org/GNOME/gnome-clocks) [GNOME](https://gitlab.gnome.org/GNOME/gnome-games) [app](https://gitlab.gnome.org/GNOME/epiphany) [uses](https://gitlab.gnome.org/World/podcasts) [libhandy](https://gitlab.gnome.org/GNOME/geary) [.](https://gitlab.gnome.org/World/Solanum)

**Comment by Kekun** [comment]
Can Glade handle incompatible versions of the same module? Understand: can Glade have *at the same time* the libhandy 0.0 and libhandy 1 catalogs installed, loading only one of these versions?

Also can Glade support extensions *right now* or will it require some development time and may not be available before a few months?

If the answer is no to any of these two, I'd maybe not have extensions until both issues are fixed, and include popular catalogs. If otherwise it's just a matter of one week of work, maybe we can wait for the extensions to be implemented.

Alternatively, if we build the catalog in the SDK, would Glade pick it up?

**Comment by bilelmoussaoui** [comment]
> Alternatively, if we build the catalog in the SDK, would Glade pick it up?

It should be, but that won't happen anytime soon

**Comment by sophie-h** [comment]
bot, build org.gnome.Glade

**Comment by sophie-h** [comment]
Can this be merged now?

**Comment by bilelmoussaoui** [comment]
I will go ahead and merge this one. Please don't revert until a proper extension support is in place :)

**Comment by albfan** [comment]
So, don't revert until you do this thing I consider is your work (it's not) and deal with people asking why flatpak has libhandy extension but their distro don't, and update libhandy dependency when new releases are there.

Please add maintainer support for anyone wanting to include whatever here (at least libhandy maintainers).

Too much people taking decisions here

Ask around how support bundle their third party extension into main app.

**Comment by xjuan** [comment]
> Can Glade handle incompatible versions of the same module? Understand: can Glade have _at the same time_ the libhandy 0.0 and libhandy 1 catalogs installed, loading only one of these versions?

It can target more than one version of the same library just like it does for gtk, but only one library can be loaded at runtime.
In practice that mean you should not remove features from libhandy, just deprecate them and set the corresponding metadata in the catalog so Glade can tell you if you are using deprecated widgets/properties etc

> Also can Glade support extensions _right now_ or will it require some development time and may not be available before a few months?
> 
> If the answer is no to any of these two, I'd maybe not have extensions until both issues are fixed, and include popular catalogs. If otherwise it's just a matter of one week of work, maybe we can wait for the extensions to be implemented.
> 
> Alternatively, if we build the catalog in the SDK, would Glade pick it up?

If you install the modules in the right place then yes, BTW webkit2gtk is not in the SDK its in glade itself
I added it in Glade because it was easier than adding it to webkit and its really low maintenance since its just a couple of objects with no UX special cases.

BTW you do not really need to have libhandy support inside flatpak, you can maker Glade load it from any other directory
See comment https://gitlab.gnome.org/GNOME/glade/-/issues/467#note_924736

**Comment by xjuan** [comment]
> I will go ahead and merge this one. Please don't revert until a proper extension support is in place :)

Sure, but you realize that this creates a confusing situation where the "Official" flatpak does not match the official tarbal.
This is something we need to resolve.
One option would be to include libhandy support in Glade itself which IIRC was never submitted or even better in Gtk (assuming its true most modern app are using it)

From the top of my head if we want to include libhandy in Glade we need
a) Icons for new libhandy widgets
b) Add libhandy widgets tests
c) Optional: Make libhandy more prominent in widget chooser

**Pull request merged**